### ISSUE TITLE
Error writing Game Shortcut bitmap

### ISSUE THREAD

**Issue by terrymacdonald** [terrymacdonald/DisplayMagician#15]
Sorry friend, I sent it by e-mail, but the attachment was not. I'm putting it again.

[DisplayMagician.log](https://github.com/terrymacdonald/DisplayMagician/files/6410518/DisplayMagician.log)
[LogReporter.20210501.083536.log](https://github.com/terrymacdonald/DisplayMagician/files/6410519/LogReporter.20210501.083536.log)

_Originally posted by @mauroartioli in https://github.com/terrymacdonald/DisplayMagician/issues/6#issuecomment-830710845_

**Comment by terrymacdonald** [comment]
Ok @mauroartioli I *think* I know where the issue could lie. I think I need to make some of the bitmap processing code more robust so that it can cope when files are missing or when files can't be accessed. The logs you sent unfortunately don't have much useful information in them. The error is coming from old code that wasn't built very well by me, so I'll go through and add extra error checking and build some robustness into it so that it can cope when things go wrong. I will also add some better logging through that part so that v1.0.7 will provide more information about the error (but still cope with it).

I will include the fix in v1.0.7 which I will release in the next 2 weeks or so.  I'm not 100% that v1.0.7 will fix your problem, but it should at least mean that DisplayMagician will stop crashing.

**Comment by terrymacdonald** [comment]
I *think* I've fixed the issue in develop branch. 1st attempt at a fix will be rolled out in 1.0.7

**Comment by terrymacdonald** [comment]
This should no longer be a problem, as I have redeveloped the icon extraction logic completely. I was never really happy with the way it worked, so I have tested and redeveloped a MUCH more robust processing mechanism. It will now almost always extract the best quality icons possible. Some of the older games still have only 32x32 icons available, but most of the newer games come with a 256x256 icons. The code will always extract the best quality icons available. 

There are a couple of corner cases where it isn't working quite correctly that I still need to troubleshoot, so I expect the new Game selection / icon extraction functionality to be released with v1.1.0 in about a week.
